### ISSUE TITLE
Refactor OutputController and store.getState().history

### ISSUE THREAD

**Issue by vcolavin** [vcolavin/nature-of-the-beast#8]
Currently the OutputController has a manifest which is of the form `{[key: uuid()]: string | JSX.Element}`. It's kind of weird for two reasons: 

1) we're storing state outside of Redux (which isn't the end of the world). We can't just put it directly in Redux though because the value might be a full-on component.

2) we're storing ordered data in an object, and relying on an array of uuids to order its values. This is suboptimal.

See here: https://github.com/reduxjs/redux/issues/1248#issuecomment-186005618

Basically, create a way of mapping some value to components (or strings).
